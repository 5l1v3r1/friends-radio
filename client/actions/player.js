export function play(media) {
  return {
    type: 'PLAY',
    payload: media
  }
}

export function pause() {
  return {
    type: 'PAUSE'
  }
}
