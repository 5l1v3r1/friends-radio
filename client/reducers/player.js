const initialState = {
  list: null,
  current: null,
  playing: false,
}

export default function playerReducer(state = initialState, action) {
  switch (action.type) {
    case 'PLAY':
      return { ...state, playing: true }
    case 'START':
      const { list, current, play } = action.payload
      return {
        ...state,
        playing: play,
        current,
        list,
      }
    case 'PAUSE':
      return { ...state, playing: false }
    case 'NEXT':
      return {
        ...state,
        current: state.current + 1
      }
    case 'PREV':
      return {
        ...state,
        current: state.current - 1
      }
    default:
      return state
  }
}
