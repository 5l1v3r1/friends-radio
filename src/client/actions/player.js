export function play() {
  return {
    type: "PLAY",
  }
}

export function pause() {
  return {
    type: "PAUSE",
  }
}

export function setCurrent(cursor) {
  return {
    type: "SET_CURRENT",
    payload: cursor,
  }
}

export function start(list, current) {
  return {
    type: "START",
    payload: {
      list,
      current,
    },
  }
}
