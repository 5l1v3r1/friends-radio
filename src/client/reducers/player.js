const initialState = {
  list: null,
  current: null,
  playing: false,
}

export default function playerReducer(state = initialState, action) {
  switch (action.type) {
    case "PLAY":
      return {...state, playing: true}
    case "START":
      const {list, current} = action.payload
      return {
        ...state,
        playing: true,
        current,
        list,
      }
    case "PAUSE":
      return {...state, playing: false}
    case "SET_CURRENT":
      return {...state, current: action.payload}
    default:
      return state
  }
}
