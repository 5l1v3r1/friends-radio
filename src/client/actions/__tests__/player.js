import {play, pause, setCurrent, start} from "../player"

test("play", () => {
  expect(play()).toMatchSnapshot()
})

test("pause", () => {
  expect(pause()).toMatchSnapshot()
})

test("setCurrent", () => {
  expect(setCurrent("cursor")).toMatchSnapshot()
})

test("start", () => {
  expect(start("foo", "bar")).toMatchSnapshot()
})
