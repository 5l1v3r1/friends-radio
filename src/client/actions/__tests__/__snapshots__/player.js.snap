// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pause 1`] = `
Object {
  "type": "PAUSE",
}
`;

exports[`play 1`] = `
Object {
  "type": "PLAY",
}
`;

exports[`setCurrent 1`] = `
Object {
  "payload": "cursor",
  "type": "SET_CURRENT",
}
`;

exports[`start 1`] = `
Object {
  "payload": Object {
    "current": "bar",
    "list": "foo",
  },
  "type": "START",
}
`;
